tests/integ/utils/sql_counter.py

```diff
@@ -388,6 +394,10 @@ def sql_count_checker(
             filter(lambda k: k[0].endswith("_count"), all_args.locals.items())
         )
     }
+    # also look into kwargs for count information
+    for (key, value) in kwargs.items():
+        if key.endswith("_count"):
+            count_kwargs.update({key: value})
```

Comment:
Why is this necessary?  I think can avoid this if add window_count=None to argument list of sql_count_checker.

Reply:
The problem is the way we treat those xxx_count = None, seems check if xxxx_count is 0, so if i add window_count=None in the argument, the tests with sql_counter_check annotation without window_count configured but have window function inside will fail.

It seems to me the one marked with explicit parameters like query_count are counter check required, but for the ones configured in kwargs are optional. At this moment, i do not want to enforce the check window function in the annotation yet.

Reply:
I think this can be fixed by adding window_count=None to sql_count_checker API and treating None count_value as an opt-out, which is the default if it is not specified.

So in sql_count_checker can do:
```    
    count_kwargs = {
        key: value
        for key, value in list(
            filter(lambda k: k[0].endswith("_count") and k[1] is not None, all_args.locals.items())
        )
    }
```

and then won't need to make this change.

I tried out and seems to work as expected if window_count not specified (or None) or window_count is not None in sql_count_checker decorator.

Reply:
So I am not sure if that is the change we want to make here, because the annotation today took bunch of _count parameter here https://github.com/snowflakedb/snowpark-python/blob/main/tests/integ/utils/sql_counter.py#L375. And when expected_count = None, we check 0.  In other words, for all annotation check, if there is an explicit declaration of _count, the count check will be enforced and 0 will be used if None. 
If i filter out all _count parameter now, that basically means for the existing test, we will no longer check the parameter anymore, i would prefer not do this to keep out check level.

Reply:
added more comments